cle/backends/elf.py

```diff
@@ -211,6 +212,13 @@ def __init__(self, binary, **kwargs):
 
         self.__parsed_reloc_tables = set()
 
+        # The linked image base should be evaluated before registering any segment or section due to
+        # the fact that elffile, used by those methods, is working only with un-based virtual addresses, but Clemories
+        # themselves are organized as a tree where each node backer internally uses relative addressing
+        self.mapped_base = self.linked_base = min(
+            map(lambda x: x['p_vaddr'],
+                filter(lambda x: x.header.p_type == 'PT_LOAD', self.reader.iter_segments())) or [0]
+        )
```

Comment:
Python's functional style is only really useful when you're using generators to prevent the whole list from ever being created. Can you break this expression into several more readable statements, like

```python
seg_addrs = [x['p_vaddr'] for x in self.reader.iter_segments() if x.header.p_type == 'PT_LOAD']
self.mapped_base = min(seg_addrs) if seg_addrs else 0
```

Also take into account sections as well, if segments are not present.